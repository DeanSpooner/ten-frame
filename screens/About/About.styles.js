import { StyleSheet } from "react-native";

export default aboutStyles = StyleSheet.create({
  text: {
    color: "#fff",
    fontSize: 20,
    textAlign: "center",
    marginTop: 15,
  },
});
