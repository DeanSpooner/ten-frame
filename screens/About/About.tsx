import React from "react";
import { Image, Text } from "react-native";
import aboutStyles from "./About.styles";

const About: React.FC<{}> = () => {
  return (
    <>
      <Image
        source={require("../../assets/images/tflogo.png")}
        style={{ width: 200, height: 200, marginBottom: 20 }}
      ></Image>
      <Text style={aboutStyles.text}>Ten Frame, made by Dean Spooner!</Text>
      <Text style={aboutStyles.text}>Built September/October 2022.</Text>
    </>
  );
};

export default About;
