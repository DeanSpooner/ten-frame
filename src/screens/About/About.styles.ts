import { StyleSheet } from "react-native";

const aboutStyles = StyleSheet.create({
  text: {
    color: "#fff",
    fontSize: 20,
    textAlign: "center",
    marginTop: 15,
  },
});

export default aboutStyles;
