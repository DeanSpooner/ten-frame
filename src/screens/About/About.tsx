import React from "react";
import { Image, Text, View } from "react-native";
import globalStyles from "../../../Global.styles";

const About: React.FC<{}> = () => {
  return (
    <View style={globalStyles.screenContainer}>
      <Image
        source={require("../../assets/images/tflogo.png")}
        style={{ width: 200, height: 200, margin: 20 }}
      ></Image>
      <Text style={globalStyles.text}>Ten Frame, made by Dean Spooner!</Text>
      <Text style={globalStyles.text}>Built September/October 2022.</Text>
    </View>
  );
};

export default About;
