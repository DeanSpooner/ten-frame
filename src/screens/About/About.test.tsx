import React from "react";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import About from "./About";

describe("<Home />", () => {
  const tree = renderer.create(<About />).toJSON() as ReactTestRendererJSON[];

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  it("has 1 child ", () => {
    expect(tree?.length).toBe(3);
  });
});
