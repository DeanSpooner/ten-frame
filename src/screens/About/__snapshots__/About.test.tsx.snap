// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
[
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/tflogo.png",
      }
    }
    style={
      {
        "height": 200,
        "marginBottom": 20,
        "width": 200,
      }
    }
  />,
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 20,
        "marginTop": 15,
        "textAlign": "center",
      }
    }
  >
    Ten Frame, made by Dean Spooner!
  </Text>,
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 20,
        "marginTop": 15,
        "textAlign": "center",
      }
    }
  >
    Built September/October 2022.
  </Text>,
]
`;
